perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -518,6 +593,148 @@ def SingleStreamThroughputBenchmark(storage_schema, host_to_connect=None):
                   host_to_connect=host_to_connect)
 
 
+def MultiStreamThroughputBenchmark(storage_schema, host_to_connect=None):
+  """A benchmark test for single stream upload and download throughput.
+
+  Args:
+    storage_schema: The schema of the storage provider to use, e.g., "gs"
+    host_to_connect: An optional host endpoint to connect to.
+
+  The object size is given by FLAGS.object_size_kb. The total number
+  of objects to transfer (summed across all streams) is given by
+  FLAGS.iterations. The benchmark starts at FLAGS.write_start_time, if
+  given, or else immediately. Reads start at FLAGS.read_start_time, if
+  given, or immediately after the writes end.
+
+  Raises:
+    LowAvailabilityError: when the storage provider has failed a high number of
+        our RW requests that exceeds a threshold (>5%), we raise this error
+        instead of collecting performance numbers from this run.
+  """
+
+  result_queue = Queue.Queue()
+
+  logging.info('About to create processes for multistream.')
+
+  processes = [threading.Thread(target=MultiStreamThroughputWorker,
+                                args=(storage_schema,
+                                      host_to_connect,
+                                      FLAGS.object_size_kb,
+                                      FLAGS.objects_per_stream,
+                                      FLAGS.write_start_time,
```

Comment:
After talking in person, I made the controlling PKB process ask for reads and writes separately. The write process sends the read process a list of already-written files by serializing it in `/tmp`, as discussed. This makes `object_storage_api_tests.py` much simpler.